### PR TITLE
Added endpoints in songs route

### LINKED ISSUE #101: Add query params in songs route
Add route to fetch `songs` by their artist, language or genre.

### Requirements
- [ ] Add route with query param `/songs?artist=xyz`
- [ ] Add route with query param `/songs?genre=xyz`
- [ ] Add route with query param `/songs?language=xyz`

### PR DESCRIPTION
**Pull request by ritik307**
# Related Issue

-   closes #101 

# Proposed Changes

- [x] Add route with query param `/songs?artist=xyz`
- [x] Add route with query param `/songs?genre=xyz`
- [x] Add route with query param `/songs?language=xyz`

# Checklist

-   [x] Tests
-   [x] Translations
-   [x] Run `ESLint`

### CODE REVIEW

**Comment by ritik307** [comment]
@ShauryaAg @paramsiddharth plz review!

File: src/api/routes/songs.ts [hunk 1]
```diff
@@ -3,10 +3,53 @@ const data = require("../../../data/songdata.json");
 
 const router = express.Router();
 
+const fetchData=(key:string,value:string)=>{
```
Comment (ShauryaAg) [inline src/api/routes/songs.ts:6]: Don't create another method. We already have a [`filter()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter) method for such things.

**Review by ShauryaAg — CHANGES_REQUESTED** [review]
Please lint the file properly before commiting

**Comment by ritik307** [comment]
image

File: src/api/routes/songs.ts (hunk 1 above)
Comment (ritik307) [reply src/api/routes/songs.ts]: @ShauryaAg Done!

**Comment by ritik307** [comment]
> 
> 
> Please lint the file properly before committing
image
Done! plz check

File: src/api/routes/songs.ts [hunk 2]
```diff
@@ -3,10 +3,52 @@ const data = require("../../../data/songdata.json");
 
 const router = express.Router();
 
+const fetchData = (key:string,value:string)=>{
```
Comment (ShauryaAg) [inline src/api/routes/songs.ts:6]: Please don't create a new function just for this. Also, use arrow functions to keep it all in one line.

**Review by ShauryaAg — CHANGES_REQUESTED** [review]

**Comment by ShauryaAg** [comment]
We can also chain multiple `.filter()` methods to query based on multiple things at once.

Something like
```js
result = data.data
          .filter(condition1) // check for genre
          .filter(condition2) // check for artist
          .filter(condition3) // check for language
```

File: src/api/routes/songs.ts (hunk 2 above)
Comment (ritik307) [reply src/api/routes/songs.ts]: Done!

**Comment by ritik307** [comment]
> 
> 
> We can also chain multiple `.filter()` methods to query based on multiple things at once.
> 
> Something like
> 
> ```js
> result = data.data
>           .filter(condition1) // check for genre
>           .filter(condition2) // check for artist
>           .filter(condition3) // check for language
> ```

I am a little confuse here chaining would pass on the output from one filter to another.How will this be useful in our case?

**Comment by ShauryaAg** [comment]
So that user may query like `/song?genre=xyz&language=abc`

Please see if you can implement this, otherwise, feel free to contact us on the discord channel.

**Comment by ShauryaAg** [comment]
> image

Do this instead
```ts
const artist:any = req.query.artist ?? null;
```

**Comment by ritik307** [comment]
image
Done!!

File: src/api/routes/songs.ts [hunk 3]
```diff
@@ -6,7 +6,26 @@ const router = express.Router();
...
+            } else {
+                const result = data.data
+                    .filter((song: any) => song.singer === (artist ?? song.singer))
+                    .filter((song: any) => song.genre === (genre ?? song.genre))
+                    .filter((song: any) => song.language === (language ?? song.language));
+
+                if (result.length) 
+                    res.status(200).json({ data: result });
+                else 
+                    res.status(404).json({ message: "No record found" });
```
Comment (ShauryaAg) [inline src/api/routes/songs.ts:24]: `res.status` should be `204`

**Review by ShauryaAg — CHANGES_REQUESTED** [review]

**Comment by ritik307** [comment]
@ShauryaAg done!

File: src/api/routes/songs.ts [hunk 4]
```diff
@@ -6,7 +6,26 @@ const router = express.Router();
 // GET request /songs
 router.get("/", (req: Request, res: Response, next: NextFunction) => {
     try {
-        res.status(200).json(data);
+        if (req.query) {
+            const artist:any = req.query.artist;
+            const genre:any = req.query.genre;
+            const language:any = req.query.language;
```
Comment (paramsiddharth) [inline src/api/routes/songs.ts:12]: The type-annotations must have a preceding space.
```ts
  const artist: any = req.query.artist;
  const genre: any = req.query.genre;
  const language: any = req.query.language;
```

**Review by paramsiddharth — CHANGES_REQUESTED** [review]

File: src/api/routes/songs.ts (hunk 4 above)
Comment (ritik307) [reply src/api/routes/songs.ts]: Done!

**Comment by ritik307** [comment]
@paramsiddharth Done! plz check

File: src/api/routes/songs.ts [hunk 5]
```diff
@@ -7,9 +7,9 @@ const router = express.Router();
 router.get("/", (req: Request, res: Response, next: NextFunction) => {
     try {
         if (req.query) {
-            const artist:any = req.query.artist;
-            const genre:any = req.query.genre;
-            const language:any = req.query.language;
+            const artist:  any = req.query.artist;
```
Comment (paramsiddharth) [inline src/api/routes/songs.ts:10]: Why 2 spaces here? 😅

**Review by paramsiddharth — CHANGES_REQUESTED** [review]

File: src/api/routes/songs.ts (hunk 5 above)
Comment (ritik307) [reply src/api/routes/songs.ts]: Done!

**Comment by ritik307** [comment]
@paramsiddharth Done! plz check

**Review by ShauryaAg — APPROVED** [review]

**Review by paramsiddharth — APPROVED** [review]

**Pull request merged**